src/goto-instrument/loop_utils.cpp

```diff
@@ -51,7 +51,7 @@ void build_havoc_code(
     goto_programt::targett t = dest.add(goto_programt::make_assignment(
       code_assignt(std::move(lhs), std::move(rhs)),
       loop_head->source_location));
-    t->code.add_source_location()=loop_head->source_location;
+    t->code_nonconst().add_source_location() = loop_head->source_location;
```

Comment:
As above: this line should be made unnecessary by fixing `goto_programt::make_assignment`.